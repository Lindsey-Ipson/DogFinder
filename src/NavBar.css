.NavBar {
    background-color: #333;
    color: white;
    overflow: hidden;
    position: fixed;
    top: 0;
    width: 100%;
    height: 40px;
    z-index: 1;
}

/* .NavBar link {
    float: right;
    display: block;
    color: white;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
    
} */

/* Unvisited link */
.NavBar a {
  color: white;
  text-decoration: none;
  margin: 10px 20px;
  display: inline-block;
}

/* Visited link */
.NavBar a:visited {
  color: white;
}

/* Mouse over link */
.NavBar a:hover {
  color: rgb(0, 208, 255);
}
