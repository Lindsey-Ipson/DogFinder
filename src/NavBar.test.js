import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from './NavBar';

const mockDogs = [
  { name: 'Dog1' },
  { name: 'Dog2' },
  // Add more mock dogs as needed
];

test('renders NavBar component with correct content', () => {
  const { getByText, getAllByRole } = render(
    <Router>
      <NavBar dogs={mockDogs} />
    </Router>
  );

  // Check if the "Home" link is rendered
  expect(getByText('Home')).toBeInTheDocument();

  // Check if each dog's name is rendered as a NavLink
  mockDogs.forEach((dog) => {
    const link = getByText(dog.name);
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', `/dogs/${dog.name.toLowerCase()}`);
  });
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <Router>
      <NavBar dogs={mockDogs} />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
