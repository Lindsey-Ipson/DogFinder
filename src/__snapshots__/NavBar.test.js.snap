// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <a
      class=""
      href="/dogs"
    >
       Home 
    </a>
    <a
      class=""
      href="/dogs/dog1"
    >
      Dog1
    </a>
    <a
      class=""
      href="/dogs/dog2"
    >
      Dog2
    </a>
  </nav>
</DocumentFragment>
`;
